x-pack/metricbeat/module/cockroachdb/status/status_test.go

```diff
@@ -0,0 +1,35 @@
+// +build !integration
+
+package status
+
+import (
+	"os"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/helper/prometheus/ptest"
+	"github.com/elastic/beats/metricbeat/mb"
+
+	// Register input module and metricset
+	_ "github.com/elastic/beats/metricbeat/module/prometheus"
+	_ "github.com/elastic/beats/metricbeat/module/prometheus/collector"
+)
+
+func init() {
+	// To be moved to some kind of helper
+	os.Setenv("BEAT_STRICT_PERMS", "false")
+	mb.Registry.SetSecondarySource(mb.NewLightModulesSource("../../../module"))
+}
+
+func TestEventMapping(t *testing.T) {
+	logp.TestingSetup()
+
+	ptest.TestMetricSet(t, "cockroachdb", "status",
```

Comment:
I wonder if this will be something we still need? Can't we just rely on the new testing framework for this? Would be great to not needed it so not golang files anymore.

Reply:
I'd like to keep all modules testable as `go test ./module/cockroachdb/...`, what I could try to use `mbtest.TestDataFiles` instead (added in  #11818).

For the future I'd like to add the possibility of testing light modules with `metricbeat` subcommands, so anyone can develop modules just with `metricbeat`. And have a helper that does the same thing when `go test` is used for the modules in the beats repo. I mention something about this [in the light modules proposal](https://docs.google.com/document/d/1R_GULET8kLOozMhHdnmn68wRgKevxCfs2S2XwVMcNXw/edit#heading=h.6n5uqbwrzc4n).

Reply:
As long as the new testing library is used, I'm good with it. We should remove ptest soon as the new library also checks fields docs etc.